simulation/common/helpers.py

```diff
@@ -584,6 +567,76 @@ def apply_race_timing_constraints(speed_kmh, start_hour, simulation_duration, ra
     return constrained_speed_kmh, not_charge
 
 
+def get_race_timing_constraints_boolean(start_hour, simulation_duration, race_type, as_seconds=True):
+    """
+
+    Applies regulation timing constraints to a speed array.
+
+    :param int start_hour: An integer representing the race's start hour
+    :param int simulation_duration: An integer representing simulation duration in seconds
+    :param str race_type: A string describing the race type. Must be one of "ASC" or "FSGP"
+    :param bool as_seconds: will return an array of seconds, or hours if set to False
+    :returns: driving_time_boolean, a boolean array with race timing constraints applied to it
+    :rtype: np.ndarray
+    :raises: ValueError is race_type is not one of "ASC" or "FSGP"
+
+    """
+
+    # (Charge from 7am-9am and 6pm-8pm) for ASC - 13 Hours of Race Day, 9 Hours of Driving
+    # (Charge from 8am-9am and 6pm-8pm) for FSGP
+
+    simulation_hours = np.arange(start_hour, start_hour + simulation_duration / (60 * 60))
+
+    if as_seconds is True:
```

Comment:
Can we change this to have a fractional input instead of a boolean? This was we can break it up by half hours, 2 mins etc.